cloud/disk_manager/internal/pkg/util/mlock.go

```diff
@@ -0,0 +1,82 @@
+package util
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+
+	"github.com/ydb-platform/nbs/cloud/tasks/errors"
+)
+
+type memoryRange struct {
+	start uintptr
+	end   uintptr
+}
+
+type procMapsItem struct {
+	memRange    memoryRange
+	permissions string
+	offset      uintptr
+	device      string
+	inode       uint64
+	pathname    string
+}
+
+func parseProcMapsLine(line string) (*procMapsItem, error) {
+	var item procMapsItem
+	n, err := fmt.Sscanf(
+		line,
+		"%x-%x %s %x %s %d %s",
+		&item.memRange.start,
+		&item.memRange.end,
+		&item.permissions,
+		&item.offset,
+		&item.device,
+		&item.inode,
+		&item.pathname,
+	)
+	if err != nil && (err != io.EOF || n < 6) {
```

Comment:
Why checking io.EOF is needed here?

Do we have tests where io.EOF happens while scanning here?

Reply:
Can we separate this if into some, and write comment about every case?